### ISSUE TITLE
Connections for archiveUserDevices are leaked

### ISSUE THREAD

**Issue by twomoose** [castle/castle-java#65]
I am getting the following log message for every call to archive a user device.

> A connection to https://api.castle.io/ was leaked. Did you forget to close a response body?

The stack trace is:
```
java.lang.Throwable: response.body().close()
	at okhttp3.internal.platform.Platform.getStackTraceForCloseable(Platform.java:143)
	at okhttp3.RealCall.captureCallStackTrace(RealCall.java:78)
	at okhttp3.RealCall.execute(RealCall.java:66)
	at io.castle.client.internal.backend.OkRestApiBackend.sendArchiveUserDevicesRequestSync(OkRestApiBackend.java:274)
	at io.castle.client.internal.CastleApiImpl.archiveUserDevices(CastleApiImpl.java:289)
```

We are using castle-java 1.5.1

**Comment by cconcannon** [comment]
@joladev I found the relevant portions of the okHttp library v 3.13.x: 

`getStackTraceForCloseable()`
https://github.com/square/okhttp/blob/okhttp_3.13.x/okhttp/src/main/java/okhttp3/internal/platform/Platform.java#L145

`captureCallStackTrace()`
https://github.com/square/okhttp/blob/okhttp_3.13.x/okhttp/src/main/java/okhttp3/RealCall.java#L88

After diving deeper into OkHttp's docs https://square.github.io/okhttp/3.x/okhttp/ I found this:
```
public abstract class ResponseBody
extends Object
implements Closeable

A one-shot stream from the origin server to the client application with the raw bytes of the response body. Each response body is supported by an active connection to the webserver. This imposes both obligations and limits on the client application.
The response body must be closed.
Each response body is backed by a limited resource like a socket (live network responses) or an open file (for cached responses). Failing to close the response body will leak resources and may ultimately cause the application to slow down or crash.

Both this class and Response implement Closeable. Closing a response simply closes its response body. If you invoke Call.execute() or implement Callback.onResponse(okhttp3.Call, okhttp3.Response) you must close this body by calling any of the following methods:

    Response.close()
    Response.body().close()
    Response.body().source().close()
    Response.body().charStream().close()
    Response.body().byteStream().close()
    Response.body().bytes()
    Response.body().string()

There is no benefit to invoking multiple close() methods for the same response body.

For synchronous calls, the easiest way to make sure a response body is closed is with a try block. With this structure the compiler inserts an implicit finally clause that calls close() for you.

   

   Call call = client.newCall(request);
   try (Response response = call.execute()) {
     ... // Use the response.
   }
 

You can use a similar block for asynchronous calls:

   

   Call call = client.newCall(request);
   call.enqueue(new Callback() {
     public void onResponse(Call call, Response response) throws IOException {
       try (ResponseBody responseBody = response.body()) {
         ... // Use the response.
       }
     }

     public void onFailure(Call call, IOException e) {
       ... // Handle the failure.
     }
   });
 

These examples will not work if you're consuming the response body on another thread. In such cases the consuming thread must call close() when it has finished reading the response body. 
```
... so yeah, we need to call one of the closing methods

**Comment by cconcannon** [comment]
The relevant file in our SDK that builds the requests to our API is `OkRestApiBackend.java`

**Comment by joladev** [comment]
We do call one of the closing methods (`.string()`)

**Comment by joladev** [comment]
@twomoose is there any other information you could share to help us debug this? We're so far unable to reproduce it. Also, it seems like the exception class is missing from your description. Not sure if it's relevant but it won't hurt :)

**Comment by joladev** [comment]
We've been unable to reproduce, but this PR might help anyway https://github.com/castle/castle-java/pull/66

**Comment by joladev** [comment]
We've released 1.5.2 with that PR. Let us know if it fixes your issue

**Comment by joladev** [comment]
Closing this for inactivity, feel free to reopen if the problem persists
